apps/routes/views/route.py

```diff
@@ -0,0 +1,72 @@
+"""
+Copyright 2020 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+
+from django.conf import settings
+from rest_framework import permissions, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from shipchain_common.mixins import SerializationType
+from shipchain_common.viewsets import ConfigurableModelViewSet, ActionConfiguration
+
+from apps.permissions import get_owner_id, owner_access_filter
+from apps.routes.models import Route
+from apps.routes.serializers import RouteSerializer, RouteCreateSerializer, RouteUpdateSerializer, RouteOrderSerializer
+
+LOG = logging.getLogger('transmission')
+
+
+class RouteViewSet(ConfigurableModelViewSet):
```

Comment:
I haven't looked at the queries for this endpoint yet but I have a feeling that some prefetch_for_includes/select_for_includes statements will go a long way here, given the complexity of RouteSerializer. Take a look at how I optimized eth.TransactionViewSet and shipments.ShipmentOverviewListView, and check out the Django docs for select_related/prefetch_related, as well as the docs here: https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#included

Reply:
I did take some time to explore that solution and the django provided prefetch options as well.  Unfortunately these appear to break the ordering provided by `order_with_respect_to`.  For now the expected number of legs on a route is low enough that performance should remain acceptable.  I've added a task to our backlog to further review this optimization in the future.